app/components/ChallengesMapView.js

```diff
@@ -71,17 +70,17 @@ export default class ChallengesMapView extends Component {
 				<View style={{flex:1}}>
 					<MapView
 						style={styles.map}
-						loadingEnabled={true}
-						showUserLocation={true}
+						loadingEnabled={false}
+						showsUserLocation={true}
 						initialRegion={this.props.initialRegion}
 					>
 					{this.props.primaryMarkers.map((marker, index) => (
-						<MapView.Marker 
+						<MapView.Marker
 							key={index}
 							identifier={marker.id}
 							coordinate={marker.latlng}
 							image={primaryMarkerImage}
-							onPress={this.onHeadMarkerPress}
+							onPress={() => this.onHeadMarkerPress(marker.id)}
```

Comment:
`this.onHeadMarkerPress` is already bound in the constructor so there is no need to bind it again. Also, I don't want only the marker.id passed as this makes me lose access to all of the attributes attached to the event (e). The attached element will be used by the wrapper function inside of this file to apply necessary animations and/or styling before calling the callback on the prop.